src/Bicep.Core/Analyzers/Linter/Rules/EnvironmentUrlHardcodedRule.cs

```diff
@@ -35,47 +35,57 @@ public override void Configure(IConfigurationRoot config)
         }
 
         protected override string FormatMessage(params object[] values)
-            => string.Format("{0} -- Found: [{1}]", this.Description, values.First());
+            => string.Format("{0} -- Found: '{1}'", this.Description, values.First());
 
         public override IEnumerable<IBicepAnalyzerDiagnostic> AnalyzeInternal(SemanticModel model)
         {
             if (this.DisallowedHosts != null && this.DisallowedHosts.Any())
             {
-                var spansToMark = new Dictionary<TextSpan, List<string>>();
-                var visitor = new Visitor(spansToMark, this.DisallowedHosts);
+                var visitor = new Visitor(this.DisallowedHosts);
                 visitor.Visit(model.SyntaxTree.ProgramSyntax);
-
-                foreach (var kvp in spansToMark)
-                {
-                    var span = kvp.Key;
-                    foreach (var hosts in kvp.Value)
-                    {
-                        yield return CreateDiagnosticForSpan(span, hosts);
-                    }
-                }
+                return visitor.DisallowedHostSpans.Select(entry => CreateDiagnosticForSpan(entry.span, entry.host));
             }
+
+            return Enumerable.Empty<IBicepAnalyzerDiagnostic>();
         }
 
         private class Visitor : SyntaxVisitor
         {
-            private readonly Dictionary<TextSpan, List<string>> hostsFound;
+            public readonly List<(string host, TextSpan span)> DisallowedHostSpans = new List<(string host, TextSpan span)>();
+
             private readonly ImmutableHashSet<string> disallowedHosts;
 
-            public Visitor(Dictionary<TextSpan, List<string>> hostsFound, ImmutableHashSet<string> disallowedHosts)
+            public Visitor(ImmutableHashSet<string> disallowedHosts)
             {
-                this.hostsFound = hostsFound;
                 this.disallowedHosts = disallowedHosts;
             }
 
             public override void VisitStringSyntax(StringSyntax syntax)
             {
-                var disallowed = syntax.SegmentValues.Where(s => this.disallowedHosts.Contains(s));
-                if (disallowed.Any())
+                var disallowedHost = syntax.SegmentValues.Select(s => findEnvironmentUrlInString(s))
+                    .Where(span => span != null)
+                    .FirstOrDefault();
+                if (disallowedHost != null)
                 {
-                    this.hostsFound[syntax.Span] = disallowed.ToList();
+                    this.DisallowedHostSpans.Add((disallowedHost, syntax.Span));
                 }
+
                 base.VisitStringSyntax(syntax);
             }
+
+            private string? findEnvironmentUrlInString(string str)
```

Comment:
>f [](http://example.com/codeflow?start=28&length=1)

nit - Method names should be capitalized.

Reply:
TypeScript habits...